test/boost/token_metadata_test.cc

```diff
@@ -41,6 +41,12 @@ namespace {
         auto strategy = seastar::make_shared<Strategy>(std::move(opts));
         return calculate_effective_replication_map(std::move(strategy), std::move(tmptr)).get0();
     }
+
```

Comment:
> The order of endpoints
returned but get_pending_endpoints/get_endpoints_for_reading
doesn't matter, the test shouldn't stick to some predefined order.
This nuance becomes important later, when we switch the
token_metadata to host_id, the order changes and the tests break.

@gusev-p I think materialized views depend on replica order. It's used to pair view replicas with base replicas. I don't know though if get_pending_endpoints/get_endpoints_for_reading are relevant in this pairing. Please investigate.

cc @nyh -- could you point to some place in the MV code which depends on the order of replicas?

Reply:
get_view_natural_endpoint()

Reply:
Right. Materialized views depends on the list of replicas for a given tablet to have the same order across time and between different nodes, but it doesn't matter for it if in different runs on different clusters, a different order is picked.

But the interaction of **pending** endpoints and materialized views was dubious, there was always at most one of them (so it didn't need any sort ordering between itself), and it didn't participate in the ordering with the usual RF replicas. So if you're asking only about **pending** endpoints, the answer probably that it doesn't matter.